diff --git a/apps/openmw/mwsound/soundmanagerimp.cpp b/apps/openmw/mwsound/soundmanagerimp.cpp
index 8471dbe0be..b28f94b676 100644
--- a/apps/openmw/mwsound/soundmanagerimp.cpp
+++ b/apps/openmw/mwsound/soundmanagerimp.cpp
@@ -22,6 +22,7 @@
 
 #include "../mwworld/cellstore.hpp"
 #include "../mwworld/esmstore.hpp"
+#include "../mwworld/class.hpp"
 
 #include "../mwmechanics/actorutil.hpp"
 
@@ -29,12 +30,14 @@
 #include "ffmpegdecoder.hpp"
 #include "openaloutput.hpp"
 #include "sound.hpp"
+#include "soundbridge.hpp"
 #include "soundbuffer.hpp"
 #include "sounddecoder.hpp"
 #include "soundoutput.hpp"
 
 namespace MWSound
 {
+    std::string g_lastResourceName;
     namespace
     {
         constexpr float sMinUpdateInterval = 1.0f / 30.0f;
@@ -263,6 +266,7 @@ namespace MWSound
             return;
 
         Log(Debug::Info) << "Playing \"" << filename << "\"";
+        MWSound::g_lastResourceName = filename.value();
 
         DecoderPtr decoder = getDecoder();
         try
@@ -325,6 +329,14 @@ namespace MWSound
         MWBase::World* world = MWBase::Environment::get().getWorld();
         const osg::Vec3f pos = world->getActorHeadTransform(ptr).getTrans();
 
+        std::ostringstream ss;
+        ss << filename.value()
+           << " | "
+           << ptr.getClass().getName(ptr);
+
+        MWSound::g_lastResourceName = ss.str();
+
+
         stopSay(ptr);
         StreamPtr sound = playVoice(std::move(decoder), pos, (ptr == MWMechanics::getPlayer()));
         if (!sound)
@@ -359,6 +371,9 @@ namespace MWSound
         if (!sound)
             return;
 
+        MWSound::g_lastResourceName =
+                std::string(filename.value()) + " | [Voice]";
+
         mActiveSaySounds.emplace(nullptr, SaySound{ nullptr, std::move(sound) });
     }
 
@@ -483,6 +498,13 @@ namespace MWSound
         Sound* result = sound.get();
         mActiveSounds[nullptr].mList.emplace_back(std::move(sound), sfx);
         mSoundBuffers.use(*sfx);
+        std::ostringstream ss;
+        ss << "file=" << sfx->getResourceName()
+           << " | [2D]";
+
+        MWSound::g_lastResourceName = ss.str();
+
+
         return result;
     }
 
@@ -566,6 +588,12 @@ namespace MWSound
             it = mActiveSounds.emplace(ptr.mRef, ActiveSound{ ptr.mCell, {} }).first;
         it->second.mList.emplace_back(std::move(sound), sfx);
         mSoundBuffers.use(*sfx);
+        std::ostringstream ss;
+        ss << "file=" << sfx->getResourceName()
+           << " ref=" << ptr.getClass().getName(ptr);
+
+        MWSound::g_lastResourceName = ss.str();
+
         return result;
     }
 
@@ -632,6 +660,9 @@ namespace MWSound
         Sound* result = sound.get();
         mActiveSounds[nullptr].mList.emplace_back(std::move(sound), sfx);
         mSoundBuffers.use(*sfx);
+        MWSound::g_lastResourceName =
+                std::string(sfx->getResourceName()) + " | [3D Pos]";
+
         return result;
     }
 
